agent/doctor/healthcheck.go

```diff
@@ -0,0 +1,70 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//      http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package doctor
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/dockerclient/dockerapi"
+	"github.com/cihub/seelog"
+)
+
+type Healthcheck interface {
+	GetHealthcheckStatus() HealthcheckStatus
+	GetHealthcheckTime() time.Time
+	RunCheck() bool
+}
+
+type dockerRuntimeHealthcheck struct {
+	// Status is the container health status
+	LatestCheckStatus HealthcheckStatus `json:"latestStatus,omitempty"`
+	// Since is the timestamp when container health status changed
+	LatestCheckTime time.Time `json:"latestStatusTime,omitempty"`
+	// Status is the container health status
+	LastCheckStatus HealthcheckStatus `json:"lastStatus,omitempty"`
+	// Since is the timestamp when container health status changed
+	LastCheckTime time.Time `json:"lastStatusTime,omitempty"`
+	client        dockerapi.DockerClient
+	lock          sync.RWMutex
+}
+
+func NewDockerRuntimeHealthcheck(client dockerapi.DockerClient) *dockerRuntimeHealthcheck {
+	return &dockerRuntimeHealthcheck{
+		client:            client,
+		LatestCheckStatus: HealthcheckStatusInitializing,
+		LatestCheckTime:   time.Now(),
```

Comment:
shouldnt these timestamps be set when the healthchecks are run in RunCheck ?

Reply:
this pr is merging into healthcheck feature branch and has a basic "demo" container runtime check.  This PR is specifically meant to add the Doctor.  We have a follow up pr to add the true container runtime check, and to set the statuses and then a third pr to poll these statuses for the message creation and send to TACS.  I'll update the description with this detail.

Reply:
this could still be set in RunCheck even if it's demo code so far couldnt it?